lbcluster/retry_module.go

```diff
@@ -0,0 +1,130 @@
+package lbcluster
+
+import (
+	"fmt"
+	"lb-experts/golbd/logger"
+	"time"
+)
+
+type Retry struct {
+	signal            chan int
+	done              chan bool
+	tick              chan bool
+	currentCount      int
+	maxCount          int
+	retryStarted      bool
+	maxDuration       time.Duration
+	retryDuration     time.Duration
+	prevRetryDuration time.Duration
+	logger            logger.Logger
+}
+
+const defaultMaxDuration = 5 * time.Minute
+
+func NewRetryModule(retryStartDuration time.Duration, logger logger.Logger) *Retry {
+	retry := &Retry{
+		maxCount:          -1,
+		currentCount:      0,
+		maxDuration:       defaultMaxDuration,
+		retryDuration:     retryStartDuration,
+		prevRetryDuration: retryStartDuration,
+		logger:            logger,
+	}
+	return retry
+}
+
+func (r *Retry) SetMaxDuration(maxDuration time.Duration) error {
+	if r.retryStarted {
+		return nil
+	}
+	if maxDuration <= 0 {
+		return fmt.Errorf("duration has to be greater than 0")
+	}
+	r.maxDuration = maxDuration
+	return nil
+}
+
+func (r *Retry) SetMaxCount(maxCount int) error {
+	if r.retryStarted {
+		return nil
+	}
+	if maxCount <= 0 {
+		return fmt.Errorf("max count has to be greater than 0")
+	}
+	r.maxCount = maxCount
+	return nil
+}
+
+func (r *Retry) start() {
+	r.retryStarted = true
+	signal := make(chan int)
+	done := make(chan bool)
+	end := time.Tick(r.maxDuration)
+	r.signal = signal
+	r.done = done
+	go func() {
+		for {
+			select {
+			case <-end:
+				r.done <- true
+				return
+			default:
+				if r.currentCount == r.maxCount {
+					r.done <- true
+					return
+				}
+			}
+		}
+	}()
+	r.run()
+}
+
+func (r *Retry) run() {
+	start := make(chan bool)
+	go func() {
+		ticker := time.NewTicker(1 * time.Minute)
```

Comment:
The ticker duration should be configurable from the `Retry` type, I think. Because, we may want to have different retry duration based on services?

Reply:
Sure. That can be configured while instantiating via the constructor itself (Line:24).